sdk/core/System.Net.ClientModel/tests/client/MapsClient/MapsClient.cs

```diff
@@ -6,52 +6,42 @@
 using System.Net.ClientModel;
 using System.Net.ClientModel.Core;
 using System.Text;
-using System.Threading;
 
 namespace Maps;
 
 public class MapsClient
 {
+    private const string LatestServiceVersion = "1.0";
+
     private readonly Uri _endpoint;
     private readonly KeyCredential _credential;
-    private readonly MessagePipeline _pipeline;
-    private readonly string _apiVersion;
+    private readonly PipelineOptions _pipelineOptions;
+    private readonly PipelinePolicy[] _clientPolicies;
 
-    public MapsClient(Uri endpoint, KeyCredential credential, MapsClientOptions options = default)
+    public MapsClient(Uri endpoint, KeyCredential credential, PipelineOptions options = default)
     {
         if (endpoint is null) throw new ArgumentNullException(nameof(endpoint));
         if (credential is null) throw new ArgumentNullException(nameof(credential));
 
-        options ??= new MapsClientOptions();
-
         _endpoint = endpoint;
         _credential = credential;
-        _apiVersion = options.Version;
 
-        if (options.PerCallPolicies is null)
-        {
-            options.PerCallPolicies = new PipelinePolicy[1];
-        }
-        else
-        {
-            var perCallPolicies = new PipelinePolicy[options.PerCallPolicies.Length + 1];
-            options.PerCallPolicies.CopyTo(perCallPolicies.AsSpan());
-        }
+        options ??= new PipelineOptions();
 
-        options.PerCallPolicies[options.PerCallPolicies.Length - 1] = new KeyCredentialAuthenticationPolicy(_credential, "subscription-key");
+        _clientPolicies = new PipelinePolicy[1];
+        _clientPolicies[0] = new KeyCredentialAuthenticationPolicy(_credential, "subscription-key");
 
-        _pipeline = MessagePipeline.Create(options);
+        // This creates the pipeline, caches it in options, and freezes options.
+        MessagePipeline.Create(options, _clientPolicies);
+
+        _pipelineOptions = options;
```

Comment:
Should this use a Copy-Constructor instead in order to let client-authors override values?

Reply:
I don't think we need/want this, since pipeline options is frozen.